sdk/program/src/pubkey.rs

```diff
@@ -170,8 +170,8 @@ pub fn bytes_are_curve_point<T: AsRef<[u8]>>(_bytes: T) -> bool {
     #[cfg(not(target_os = "solana"))]
     {
         curve25519_dalek::edwards::CompressedEdwardsY::from_slice(_bytes.as_ref())
-            .decompress()
-            .is_some()
+            .map(|compressed_edwards_y| compressed_edwards_y.decompress().is_some())
+            .unwrap_or(false)
```

Comment:
what do you mean "putting false result into the Ok part of the Result value"? afaik the original version of code will only return `true` or `false`.

Reply:
If you follow the logic of the expression, you'll see the following:
```rust
CompressedEdwardsY::from_slice(_bytes.as_ref())
```
produces `Result<CompressedEdwardsY, Error>` - here the "good" value is in the `Ok()` part of the result, and the error is in `Err()`.
This is expected.

But after
```rust
    .map(|compressed_edwards_y| compressed_edwards_y.decompress().is_some())
```
you get `Result<Bool, Error>`.
Where `Ok(true)` indicates a successful decompression, while `Ok(false)` indicates a decompression failure.  With `Err(_)` still holding the parsing error.
This is somewhat confusing, as the failed decompression is represented by an `Ok(_)` value, even while we still have an `Err(_)`.

Finally
```rust
    .unwrap_or(false)
```
will combine the parsing error and the decompression failure into a single `false`.

The whole thing works.  But when I try to follow it, it just feels like following a puzzle, rather than reading something that is supposed to explain an intent of the original author.

Reply:
okay! thank you!